### PR TITLE
 scripts/websitepreprocess.sh,docs/*: Fix broken links

### PR DESCRIPTION
**Pull request by LiliC**
## Changes:

- Up to now, all the relative links were replaced in the `scripts/websitepreprocess.sh` by generating the URL for the website to correctly parse the links, that always added the `/docs/` dir in the path, which did not work for some of them (e.g. Makefile link below) as they were located outside of that directory. 
- Added small doc note on how to run the website locally, let me know if that is useful? Saves a small amount of time for not needing to go look at the Makefile :)
- Some other small cleanups.
- The `go.sum` change was automatic when I ran `go mod vendor`, but can remove it if you are getting a different result. 🤷‍♀ 

## Verification

Tested all the links and they resolve both in markdown on github and by running the website locally. 

As a side note/question, there are some tools out there that can be added as part of the CI process to test the links and make sure they are valid.

---
p.s. Great docs! <3

### CODE REVIEW

File: docs/contributing/how-to-contribute-to-docs.md [hunk 1]
```diff
@@ -94,8 +94,22 @@ Every PR is building website and on success it shows the link to preview.
 
 We use [Netlify](https://www.netlify.com/) for hosting. We are using Open Source license (PRO). Thanks Netlify for this!
 
-On every commit to `master` netlify runs CI that invokes `make web` (defined in [netlify.toml](https://github.com/improbable-eng/thanos/blob/master/netlify.toml)))
```
Comment (bwplotka) [inline docs/contributing/how-to-contribute-to-docs.md]: We need this prefix IMO (similar to any resource outside of `docs`)

File: docs/contributing/how-to-contribute-to-docs.md [hunk 2]
```diff
@@ -94,8 +94,22 @@ Every PR is building website and on success it shows the link to preview.
...
+On every commit to `master` netlify runs CI that invokes `make web` (defined in [netlify.toml](/netlify.toml))
 
 NOTE: Check for status badge in README for build status on the page.
 
 If master build for netlify succeed, the new content is published automatically.
+
+## Test locally
```
Comment (bwplotka) [inline docs/contributing/how-to-contribute-to-docs.md]: I think this makes sense to have, but can we somehow integrate this into `Testing` section above? (:

File: docs/contributing/how-to-contribute-to-docs.md [hunk 3]
```diff
@@ -94,8 +94,22 @@ Every PR is building website and on success it shows the link to preview.
...
+
+To test any changes to the docs locally, just run the following command to build the documention website:
+
+```bash
+make web
+```
+
+To start serving the website locally run the following command and you will be able to access the website on `localhost:1313` by default:
```
Comment (bwplotka) [inline docs/contributing/how-to-contribute-to-docs.md]: Worth to add that you don't need to run `make web` if you do `make web-serve`? Easy to be confused by reading this and think that those two steps are essential, not sure.

File: docs/getting-started.md [hunk 4]
```diff
@@ -123,8 +121,8 @@ thanos sidecar \
     --grpc-address              0.0.0.0:19090              # GRPC endpoint for StoreAPI
 ```
 
-* _[Example Kubernetes manifest](https://github.com/improbable-eng/thanos/tree/master/tutorials/kubernetes-demo/manifests/prometheus-ha-sidecar.yaml)_
```
Comment (bwplotka) [inline docs/getting-started.md]: Same issue as everywhere else when using outside `docs` resources ):

File: scripts/websitepreprocess.sh [hunk 5]
```diff
@@ -65,9 +65,9 @@ EOT
 done
 
 # 3. All the absolute links needs are directly linking github with the given commit.
```
Comment (bwplotka) [inline scripts/websitepreprocess.sh]: typo in comment ): I know it's not yours, but can we reword?

File: docs/contributing/how-to-contribute-to-docs.md (hunk 1 above)
Comment (LiliC) [reply docs/contributing/how-to-contribute-to-docs.md]: Where does this not work for you? I could be missing something obvious, sorry for that! :D 

On the website itself, this works fine -> https://deploy-preview-1176--thanos-io.netlify.com/contributing/how-to-contribute-to-docs.md/ 
and in markdown https://github.com/improbable-eng/thanos/blob/6a5b9b41856fa8781f3bb352322422cc25538aa2/docs/contributing/how-to-contribute-to-docs.md#deployment

**Comment by LiliC** [comment]
> so I am bit surprise the broken links you mentioned were not caught up. Which one were not working?

It all started with the Makefile link here -> https://thanos.io/getting-started.md/#prometheus

File: docs/getting-started.md (hunk 4 above)
Comment (bwplotka) [reply docs/getting-started.md]: ignore me you just fixed the processing script as well (:

File: docs/contributing/how-to-contribute-to-docs.md (hunk 1 above)
Comment (bwplotka) [reply docs/contributing/how-to-contribute-to-docs.md]: ignore me you just fixed the processing script as well (: Thanks ;p That's way more consistent :+1:

File: docs/components/query.md [hunk 6]
```diff
@@ -66,19 +66,19 @@ However, for additional Thanos features, Thanos, on top of Prometheus adds
 
 ### Partial Response
 
-QueryAPI and StoreAPI has additional behaviour controlled via query parameter called [PartialResponseStrategy](../../pkg/store/storepb/rpc.pb.go).
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply docs/components/query.md]: ignore me you just fixed the processing script as well (:

**Comment by bwplotka** [comment]
> so I am bit surprise the broken links you mentioned were not caught up. Which one were not working?

I think the key part is we test links for markdown not from website perspective (: We may want to add that one in the future, some issue would be nice for this (:

**Comment by LiliC** [comment]
> We may want to add that one in the future, some issue would be nice for this (:

Opened an issue for that #1177 :) 

Will address the other comments later, thanks for your quick response!

File: docs/contributing/how-to-contribute-to-docs.md [hunk 7]
```diff
@@ -26,17 +26,16 @@ type: ...
...
 ## Links
 
 Aim is to match linking behavior in website being THE SAME as Github. This means:
 
 * For files in Hugo <content> dir (so `./docs`). Put `slug: /<filename with extension>`
-* For any sub dir add to `website/hugo.yaml` new dir as key  `permalinks:` with `<dir>: /<dir>/:filname.md`
+* For any sub dir add to `website/hugo.yaml` new dir as key `permalinks:` with `<dir>: /<dir>/:filename.md`
 
-Then everywhere use native markdown *relative* symbolic links if you want to reference some mardkown file from `docs`:
+Then everywhere use native markdown *relative* symbolic links if you want to reference some markdown file from `docs`:
```
[the remark this answers is not in the document]
Comment (LiliC) [reply docs/contributing/how-to-contribute-to-docs.md]: Agreed, I would say we keep just the "absolute" part and remove this line altogether. Think this makes sense, so the next line which reads:

> Or absolute path to the project repository if you want to link to exact commit e.g:
> `[title]( /Makefile )`

SGTY?

File: docs/contributing/how-to-contribute-to-docs.md (hunk 3 above)
Comment (LiliC) [reply docs/contributing/how-to-contribute-to-docs.md]: Yeah, that makes sense, sorry docs are hard to write :D I rephrased it a bit, just kept the `make web-serve` as the other one is explained further down anyways.

File: scripts/websitepreprocess.sh (hunk 5 above)
Comment (LiliC) [reply scripts/websitepreprocess.sh]: Reworded, hopefully, makes sense.

**Comment by LiliC** [comment]
@bwplotka done with one question, PTAL, thanks!

File: docs/contributing/how-to-contribute-to-docs.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (bwplotka) [reply docs/contributing/how-to-contribute-to-docs.md]: :+1:

[the remark this answers is not in the document]
Comment (bwplotka) [reply docs/contributing/how-to-contribute-to-docs.md]: I guess this is the last bit to change before merge (:

[the remark this answers is not in the document]
Comment (LiliC) [reply docs/contributing/how-to-contribute-to-docs.md]: @bwplotka Done :) ✔️

**Pull request merged**